lib/puppet/indirector/request.rb

```diff
@@ -217,11 +217,11 @@ def do_request(srv_service=:puppet, default_server=nil, default_port=nil, &block
       if primary_server = Puppet.settings[:server_list][0]
         bound_port = primary_server[1]
       else
-        bound_port = nil
+        bound_port = Puppet.settings[:masterport]
       end
     end
-    self.server = default_server || bound_server || Puppet.settings[:server]
-    self.port   = default_port || bound_port || Puppet.settings[:masterport]
+    self.server = default_server || bound_server
+    self.port   = default_port || bound_port
```

Comment:
I'm having a hard time groking how this change is functionally different. It still seems, at least with a superficial understanding of the flow here, that we're going to fall back to the value of `Puppet.settings[:server]` for `self.server` if we fail on `Puppet.lookup(:server)` and the `server_list` setting is not populated..